### ISSUE TITLE
[a11y]: Switcheritem tag giving accessibility error

### ISSUE THREAD

**Issue by Mahalakshmivignesh** [carbon-design-system/carbon#16772]
### Package

@carbon/react

### Browser

Chrome

### Package version

1.58.1

### Automated testing tool and ruleset

IBM Equal accessibility checker

### Description

Hi all, <Switcheritem> element with only aria-label is giving accessibility error . the corresponding <a> tag generated by the carbon has role as generic.  Error - The ARIA attributes “aria-label” are not valid for the element <a> with implicit ARIA role “generic” . The switcher item i am using is without href link. Hence, i believe, the role is assigned as generic. I tried to assign the role explicitly for the switcher item. but it is not an accepted property of the switcher item tag. 

### Reproduction/example

Unable to recreate the issue with the template

### Steps to reproduce

Declare a header container with header panel code sample given below:
					

```
<HeaderPanel
						expanded={isSideNavExpanded}
						href="#test"
					>
						<Switcher aria-label="Switcher Container">
							<SwitcherItem aria-label="user">
								"user01"
							</SwitcherItem>
						</Switcher>
					</HeaderPanel>
```

### Suggested Severity

None

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by alisonjoseph** [comment]
Looks like the issue you are seeing is because `aria-label` doesn't match the link text.
`
<SwitcherItem aria-label="these values needs to match">these values needs to match</SwitcherItem>`

**Comment by Mahalakshmivignesh** [comment]
Hi, this is not resolving the issue when the switcher tag doesnt have href link. From the error it seems that the <a> tag generated for the switcher item is expecting a role. but when there is no href, the role is generic. for a generic role, aria-label cant be added. so we shud 
1. able to add role for the <a> tag generated by switcher tag. or 
2. able to remove the aria-label from the switcher tag when href is not provided

**Comment by tay1orjones** [comment]
@2nikhiltom would you be able to take a look at this? You're probably most familiar due to your work on https://github.com/carbon-design-system/carbon/pull/16043

**Comment by 2nikhiltom** [comment]
Hey ! I have two observations 

1 ) It is evident from the docs that a `SwitcherItem` is always expected to have a `link/href` to redirect somewhere when clicked

<img width="950" alt="image" src="https://github.com/carbon-design-system/carbon/assets/63502271/75612119-4738-42ae-b20d-41c6fb46ffd9">

https://carbondesignsystem.com/components/UI-shell-right-panel/usage/
<img width="1009" alt="image" src="https://github.com/carbon-design-system/carbon/assets/63502271/6df841c8-9234-4dad-8e37-09b53f8a8057">




@Mahalakshmivignesh  : Could you please help us understand the useCase of code you shared 
i.e `SwitcherItem without a href `
```
<HeaderPanel
						expanded={isSideNavExpanded}
						href="#test"
					>
						<Switcher aria-label="Switcher Container">
							<SwitcherItem aria-label="user">
								"user01"
							</SwitcherItem>
						</Switcher>
					</HeaderPanel>
```




2) Also I feel we can add a role='link' to the `SwitcherItem` component 
<img width="537" alt="image" src="https://github.com/carbon-design-system/carbon/assets/63502271/1d736105-792e-4802-9627-2eb8f564cede">

**Comment by Mahalakshmivignesh** [comment]
@2nikhiltom - In my case, it is. display only field & there is no link associated with it. On clicking this button it shud display the username and there is no link to that

**Comment by Mahalakshmivignesh** [comment]
Error screenshot - 
<img width="1300" alt="image" src="https://github.com/carbon-design-system/carbon/assets/141624666/37727f1b-5c9a-4df1-bf9a-b2f906be8473">

**Comment by Mahalakshmivignesh** [comment]
the href attribute is optional for the switcher tag. If in any case, we don't supply the href, the role not getting assigned and i also cant add the role attribute to the switcheritem/switcher tag.

**Comment by 2nikhiltom** [comment]
Hi @Mahalakshmivignesh ,

1) SwitcherItem is always expected to have a link (href) to redirect somewhere when clicked. 
In your example, it displays a username without a link. A workaround is to pass an empty link (#), I personally would not recommended it but it bypasses the accessibility error

- Also under the hood,  SwitcherItem uses PolymorphicProps logic, and the `as` prop allows specifying any HTML element or React component to render the Link component

- In your case, you could use something like this 

` <SwitcherItem as={React.node}>{user.name}</SwitcherItem>`

-  Here is an [stackblitz_example ](https://stackblitz.com/edit/github-b2de36?file=src%2FApp.jsx)

 
2) Something like [Profile menu ](https://pages.github.ibm.com/cdai-design/pal/patterns/profile-header-menu/usage/) could also be used instead 

@tay1orjones | @laurenmrice Any thoughts on this ?

**Comment by Mahalakshmivignesh** [comment]
as the workaround provided works for me, i am closing this ticket.
